test/Base_Tests/src/Data/Vector_Spec.enso

```diff
@@ -1053,14 +1054,49 @@ add_specs suite_builder =
                 builder.append 1
             v3 . should_equal [1]
 
-        group_builder.specify "Should propagate dataflow errors" <|
+        group_builder.specify "Should propagate uncaught dataflow errors returned from build callback" <|
             v = Vector.build builder->
                 builder.append 1
                 builder.append 2
                 Error.throw (Illegal_Argument.Error "asdf")
             v . should_fail_with (Illegal_Argument.Error "asdf")
 
-        group_builder.specify "example" <|
+        group_builder.specify "Builder should propagate an uncaught dataflow error if it is appended with .append" <|
+            v = Vector.build builder->
+                builder.append 1
+                builder.append (Error.throw (Illegal_Argument.Error "asdf"))
+                builder.append 3
```

Comment:
```suggestion
                Test.fail "should not reach here because builder is immediately aborted on error"
                builder.append 3
```

Reply:
We could also show here in the tests that the build operation is interrupted upon the error.